doc/rule-proposal.md

```diff
@@ -0,0 +1,84 @@
+# Proposing Axe-core Rules
+
+This document outlines the process of proposing a rule. For a technical description on how to build a rule, read [Developing Axe-core Rules](./rule-development.md)
+
+Before you start coding a new rule for axe-core, you *must* create a Github issue to document the rule you want to create. There are many considerations to writing a good rule. It must have no false positives, which is difficult, because there are always many many edge cases to consider. Additionally, a known problem of accessibility testing, is that not all testers hold the same interpretation of the accessibility guidelines. All rules *must* be consistent with the interpretation of Deque Systems, the developer behind Axe-core. 
+
+In addition to giving the axe-core development team an opportunity to provide feedback on the proposed rule, the Github Issue will serve as documentation of that rule for the future.
+
+## Rules Format
+
+All Github issues that propose a rule must be tagged as *rule*, and must use the following format:
+
+### Intro
+
+In one sentence describe what the rule does.
+
+Example: "Ensures ARIA attributes are allowed for an element's role"\
+
+### Tags
+
+Indicate which tags the rule should use.
+
+Example: wcag2a, wcag211, cat.keyboard
+
+### Selector
+
+If possible using a CSS selector, otherwise describe in one sentence using plain language what elements the rule selects.
+
+Example 1: `input[type=checkbox][name]`
+
+Example 2: Select each node that has an attribute starting with `aria-`.
+
+### Checks:
+
+Make each check a subheading of `checks`. Give the check name in the heading, and indicate if the check type is `any`, `all` or `none`.
+
+In short sentences, using plain language, describe what conditions will lead to the check returning false, true or undefined. Keep the steps simple and short. You don't have to write out all the logic. Preferably not, just give the high level view of what the check does.
+
+**Example 1:**
+
+`###` aria/allow-attr (any)
+
+1. Look up the element role
+2. Look up a list of aria attributes allowed for that role
+3. Return false if the element has aria attributes not in the list
+4. Otherwise return true
+
+**Example 2:**
+
+`###` keyboard/focusable-no-name (none)
+
+1. If the element is not focusable, return false
+2. If the element has an accessible name, return false
+3. Otherwise return true
+
+## Template
```

Comment:
The template is missing a section for specifying tags, which you ask the submitter to supply in the section above.